File: 8kyu/FilteringEvenNumbers1.cs
/*
 * Filtering even numbers (Bug fixes) - C#
 *
 * The method is supposed to remove even numbers from the list and return a
 * list that contains odd number. However, there is a bug in the method that
 * needs to be resolved.
 *
 * Author: Unai de la O
 */
 using System;
 using System.Collections.Generic;
 using System.Linq;

 public class FilteringEvenNumbers1
 {
   public static List<int> FilterOddNumber(List<int> listOfNumbers)
   {
     for (int i = 0; i < listOfNumbers.Count; i++)
     {
       if (listOfNumbers[i]%2 == 0)
       {
         listOfNumbers.RemoveAt(i);
         i--;
       }
     }
     return listOfNumbers;
   }
 }

File: 8kyu/FilteringEvenNumbers1.java
/**
 * Filtering even numbers (Bug fixes) - Java
 *
 * The method is supposed to remove even numbers from the list and return a
 * list that contains odd number. However, there is a bug in the method that
 * needs to be resolved.
 *
 * @author Unai de la O
 */

import java.util.List;

public class FilteringEvenNumbers1 {
  public static List<Integer> filterOddNumber(List<Integer> listOfNumbers)
  {
    for (int i = 0; i < listOfNumbers.size(); i++)
    {
      if (listOfNumbers.get(i)%2 == 0)
      {
        listOfNumbers.remove(i);
        i--;
      }
    }
    return listOfNumbers;
  }
}
